### DISCUSSION TITLE
if number length > 15 Error number returned

### DISCUSSION THREAD

**Discussion by thankdepend** [sidorares/node-mysql2#1476] [Q&A] [answered]
select * from hrms_user_education_info where user_id = "897186557948076033" and company_id = "101"
---------------------
result [
  TextRow {
    id: 897186557948076000,
    user_id: 897186557948076000,
    user_name: null,
    company_id: 101,
    school_name: null,
    education: null,
    start_date: null,
    end_date: null,
    major: null,
    is_degree: null,
    education_path: null,
    degree_path: null,
    remark: null,
    status: 'ACTIVE',
    is_delete: 0,
    record_version: 1,
    create_date: 2021-12-21T06:45:03.000Z,
    create_user_code: '21032391',
    create_user_name: 'hrms001',
    last_upd_date: 2021-12-21T06:45:03.000Z,
    last_upd_user_code: '21032391',
    last_upd_user_name: 'hrms001',
    orderby: null
  }
]

long number  The accuracy is inaccurate

**Comment by sidorares** [comment, marked as answer]
This is likely expected behaviour, we map numeric types to a JS number and you can't have precise integer larger than `Number.MAX_SAFE_INTEGER` - see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER

In your case possible solutions are:
- don't use number for user_id and use string type instead. Note that there are lots of other ways large integer id can bite you ( for example, you'll likely have data corruptions serialising / deserialising id to a JSON )
- pass `supportBigNumbers` option:

> Enabling both supportBigNumbers and bigNumberStrings forces big numbers (BIGINT and DECIMAL columns) to be always returned as JavaScript String objects (Default: false). Enabling supportBigNumbers but leaving bigNumberStrings disabled will return big numbers as String objects only when they cannot be accurately represented with [JavaScript Number objects] (http://ecma262-5.com/ELS5_HTML.htm#Section_8.5) (which happens when they exceed the [-2^53, +2^53] range), otherwise they will be returned as Number objects. This option is ignored if supportBigNumbers is disabled.
